### ISSUE TITLE
Login through SSO keeps sending "We've noticed a new login..." from the same device

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#16919]
**Describe the bug**
Login through SSO keeps sending "We've noticed a new login..." from the same device. This is annoying and means the functionality is disabled completely as a "fix" by using the environment variable `MB_SEND_EMAIL_ON_FIRST_LOGIN_FROM_NEW_DEVICE=FALSE`

**To Reproduce**
Setup an instance with SSO (Google, SAML, etc) and login thru SSO, which sends a notification, then logout and login again, which sends a notification again.
Some setups requires re-login, when the browser is closed, which means that users might see this email several times per day.

**Information about your Metabase Installation:**
Tested 1.39.4 and 1.40.0-rc2

**Comment by bchrobot** [comment]
We just updated from v0.38.1 to v0.40.1 and are now experiencing this for API calls using basic username/password auth.

**Comment by noahmoss** [comment]
@flamber I can't seem to repro this locally on the latest master, or on stats. Are there any other details that might help to trigger this?

**Comment by flamber** [comment]
@noahmoss I have a feeling that the users were initially created in Metabase (meaning they're created as email logins) and then used as Google/SAML logins. I can spin up a setup to fully validate it, but it was reported by a couple of EE, which are older setups, so I'm fairly sure the users were created in Metabase before they switched to SSO.

@bchrobot You are looking for #16264, which is unrelated to this issue. Same workaround though.

**Comment by noahmoss** [comment]
@flamber I'm trying to repro this again, and am still not able to.

Here's what I tried, on a fresh install:
1. Created a non-admin user, with a Gmail email address
2. Set up the password for the non-admin user and logged in
3. Verified that a "We've noticed a new login" email is sent
4. Logged out, then logged back in as an admin
5. Set up Google sign-in
6. Logged out as admin
7. Logged in as the non-admin user via Google sign in
8. Did _not_ receive another "We've noticed a new login" email
